azure/functions/decorators/core.py

```diff
@@ -73,17 +73,19 @@ class Binding(ABC):
     attribute in function.json when new binding classes are created.
     Ref: https://aka.ms/azure-function-binding-http """
 
+    EXCLUDED_INIT_PARAMS = {'self', 'kwargs', 'type', 'data_type', 'direction'}
+
     @staticmethod
     @abstractmethod
     def get_binding_name() -> str:
         pass
 
     def __init__(self, name: str,
                  direction: BindingDirection,
-                 is_trigger: bool,
-                 data_type: Optional[DataType] = None):
-        self.type = self.get_binding_name()
-        self.is_trigger = is_trigger
+                 data_type: Optional[DataType] = None,
+                 type: Optional[str] = None):  # NoQa
+        self.type = self.get_binding_name() \
+            if self.get_binding_name() is not None else type
```

Comment:
This can be None for both cases - right? As type is optional. Why not utilize the class name or some other property in that case?

Reply:
The generic decorators take care of type - to ensure the type is populated. And in the native-binding case, the get_binding_name is always implemented.

discussed offline; No need to make any changes